packages/dashboard-core-plugins/src/panels/IrisGridPanel.jsx

```diff
@@ -681,12 +669,29 @@ export class IrisGridPanel extends PureComponent {
 
   loadPanelState(model) {
     const { panelState } = this.props;
+    const { irisGridStateOverrides } = this.state;
     if (panelState == null) {
```

Comment:
This check causes it to return if you create a new panel and add filters and save them to the filter set manager without closing/re-opening the dashboard first. That's because the `panelState` provided in the props is undefined.

I think you should check both `panelState == null && Object.keys(irisGridStateOverrides).length === 0` and return then, then if `panelState == null`, use `panelState` from `this.state` instead.